contracts/svm/programs/express_relay/src/swap.rs

```diff
@@ -96,6 +95,24 @@ impl<'info> Swap<'info> {
         )?;
         Ok(())
     }
+
+    pub fn check_enough_balances(&self, args: &SwapArgs) -> Result<()> {
+        require_gte!(
+            self.searcher_ta_mint_searcher.amount,
+            args.amount_searcher,
+            ErrorCode::SearcherInsufficientBalance
+        );
+        // We want the program to never fail after the check below.
```

Comment:
in particular we dont token transfers to fail after due to wrong delegate or other issues. maybe worth having this comment in the call to `check_enough_balances` in the swap instruction?